xamarin/Xamarin.Forms
# Fix 8743 - now using specific style in SearchBar [UWP]

### Description of Change ###

The UWP SearchBarRenderer relied on a default style for its AutoSuggestBox. However, this default was overwritten by [these lines](https://github.com/xamarin/Xamarin.Forms/blob/f51099c3823cd7433d1fa95c4cff9227b2a59629/Xamarin.Forms.Platform.UAP/Forms.cs#L37-L44) in #7319. With the wrong style, setting the FontSize had no effect.

These changes amend the style so that is has an x.Key, and specifically sets this style in _SearchBarRenderer_; also changes the filename for the style's XAML file to make it clearer it is a local style.

### Issues Resolved ### 

- fixes #8743

### API Changes ###
 
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###

SearchBar is now using the intended style - changes to, for instance, FontSize will be respected.

### Before/After Screenshots ### 
Before:
![fix8743-before](https://user-images.githubusercontent.com/3807144/70293892-06104a00-17e2-11ea-9bff-970395389f95.png)

After:
![fix8743-after](https://user-images.githubusercontent.com/3807144/70293902-0e688500-17e2-11ea-8222-9023ab60225f.png)


### Testing Procedure ###

On UWP, go to test case 8743. Check that the three Search Bars have correct sizes.


### Notes ###

The UWP _ShellSectionRenderer_ also uses an AutoSuggestBox. This does not seem to have properties specifiyng the looks of the Control, however, and is probably happy about using the UWP default. Therefore I haven done anything to change the style it is using. If it turns out the style _should_ be specified, then [this is probably where](https://github.com/xamarin/Xamarin.Forms/blob/0250aefe57d3d1e9b9cce80ecd3a9104a33383ce/Xamarin.Forms.Platform.UAP/Shell/ShellSectionRenderer.cs#L28) it should be done.

### PR Checklist ###
